util/src/main/java/org/killbill/billing/util/tag/dao/TagDefinitionModelDao.java

```diff
@@ -31,6 +31,13 @@
 
 public class TagDefinitionModelDao extends EntityModelDaoBase implements EntityModelDao<TagDefinition> {
 
+    /**
+     * FIXME-1615 : We can use String Joiner here, but:
+     * 1. it only accept string, so we need to do things like
+     *    tag.getApplicableObjectTypes().stream().map( ... ).collect() (see line 58)
+     * 2. as this is in several places, maybe we need a utility classes dedicated for this
+     * 3. And lead to, "Can't we just use Google's Joiner?")
```

Comment:
Indeed, good question.

The primary reason for this work is to follow the migration suggestions from https://guava.dev/releases/23.0/api/docs/com/google/common/collect/Iterables.html:

> Java 8 users: several common uses for this class are now more comprehensively addressed by the new Stream library. Read the method documentation below for comparisons. This class is not being deprecated, but we gently encourage you to migrate to streams.

A secondary goal is to investigate if we can get rid of the library completely, mostly because of transitive dependency issues (check the comments from https://github.com/google/guava/issues/3320). I don't know if it's really feasable or realistic though. I would suggest you keep track of these questions and we make a decision down the line (if it's only a couple of utility functions, maybe then it's worth copying them).

Reply:
Ok so lets keep the comment for a while.